### PR TITLE
organizing the tracks in the ipscan results

### PR DESCRIPTION
**Pull request by gustavo-salazar**
The InterProScan results include a protein viewer, this one however was displaying some categories in a different way than the viewer in the protein pages.

Some of the inconsistencies were:
* The category called Predictions should be Other features
* Funfam where displayed as Unintegrated features
* PIRSR residues were displayed as features.
* Residues were displayed together with their associated entries. However, in the protein viewers, those are displayed in a separate category.

### CODE REVIEW

File: src/components/IPScan/Summary/serializers.js [hunk 1]
```diff
@@ -12,6 +12,9 @@ type Location = {
 }
 */
 
+const OTHER_FEATURES_DBS = ['funfam'];
```
Comment (typhainepl) [inline src/components/IPScan/Summary/serializers.js:15]: Does InterProScan reports Pfam-N matches? If so I think Pfam-N should also be added to that list.

File: src/components/IPScan/Summary/serializers.js [hunk 2]
```diff
@@ -95,12 +118,20 @@ const condenseLocations = (
...
+    residues: any[],
+    other_residues: any[],
+  } */ = {
+    unintegrated: [],
+    other_features: [],
+    residues: [],
+    other_residues: [],
+  };
   const unintegrated = {};
+  const predictions = {};
```
Comment (typhainepl) [inline src/components/IPScan/Summary/serializers.js:133]: I would suggest to change `predictions` to `other_features` for consistency.

File: src/components/IPScan/Summary/serializers.js [hunk 3]
```diff
@@ -121,26 +152,28 @@ export const mergeData = (matches, sequenceLength) => {
...
+        ...processedMatch,
+        accession: `residue:${processedMatch.accession}`,
+        residues,
+      });
+    }
     if (NOT_MEMBER_DBS.has(library)) {
-      processedMatch.accession += ` (${mergedData.predictions.length + 1})`;
       processedMatch.source_database = library; // Making sure the change matches the ignore list.
-      mergedData.predictions.push(processedMatch);
+      if (processedMatch.accession in predictions) {
```
Comment (typhainepl) [inline src/components/IPScan/Summary/serializers.js:170]: same comment as above, regarding the `predictions` variable name

**Review by typhainepl — COMMENTED** [review]
The changes looks good overall! A few comments below, I've been searching for the UniProt [P00519](https://www.ebi.ac.uk/interpro/protein/UniProt/P00519/) sequence.

Regarding PIRSR, in the protein viewer of the protein page, the category is shown (e.g. ACT_SITE: Proton acceptor. NP_BIND: ATP. BINDING: ATP.), would it be possible to do the same for the InterProScan search results?

In the protein viewer of the protein page, a logo is shown before the InterPro entry giving information about the type, could you do the same here?

**Comment by gustavo-salazar** [comment]
OK changes done!

**Comment by typhainepl** [comment]
Looking good to me.
Once Matthias has confirmed the info about Pfam-N, you can merge.

File: src/components/IPScan/Summary/serializers.js (hunk 1 above)
Comment (matthiasblum) [reply src/components/IPScan/Summary/serializers.js:15]: Correct. Pfam-N matches were provided by Google Research and are not reported by InterProScan.

File: src/components/IPScan/Summary/serializers.js [hunk 4]
```diff
@@ -121,26 +155,28 @@ export const mergeData = (matches, sequenceLength) => {
...
-              accession: match.signature.accession,
-              locations: sites.map((site) => ({
-                description: site.description,
-                fragments: site.siteLocations,
-              })),
-            }
-          : null,
-      )
-      .filter(Boolean);
-    if (residues.length > 0) processedMatch.residues = residues;
```
Comment (laisecf) [inline src/components/IPScan/Summary/serializers.js:138]: A doubt ^^ what did that line do exactly? Why this is not necessary anymore after changes?

**Review by laisecf — APPROVED** [review]

File: src/components/IPScan/Summary/serializers.js (hunk 4 above)
Comment (gustavo-salazar) [reply src/components/IPScan/Summary/serializers.js:138]: The changes were calculated in the variable `residues`, however at the end what we return is the variable `processedMatch` so the line is just putting the calculation in the output variable when it found any residues.
This, in the new code, has been changed to line 165, where the residues are now included together with `processedMatch` to get the effect that the residues are part of the entry

Comment (laisecf) [reply src/components/IPScan/Summary/serializers.js:138]: ok!! thanks 🤗

**Pull request merged**